### ISSUE TITLE
Miscategorises URLs as comments

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#339]
**What version of TypeScript are you using?**

2.4

**What version of `typescript-eslint-parser` are you using?**

`ts-2.4` branch

**What code were you trying to parse?**

```ts
const link = <a href="example.com">http://example.com</a>
```

**What did you expect to happen?**

`//example.com</a>` is not a comment

**What happened?**

```json
  "comments": [
    {
      "type": "Line",
      "value": "example.com</a>",
      "range": [
        40,
        57
      ],
      "loc": {
        "start": {
          "line": 1,
          "column": 40
        },
        "end": {
          "line": 1,
          "column": 57
        }
      }
    }
```

**Comment by soda0289** [comment]
I think we will need to change how comments are processed inside JSX. We should only accept multiline comments inside JSX literal `{/* Valid comment */} `

**Comment by azz** [comment]
Looks like this one is still causing some issues.

**Comment by kaicataldo** [comment]
Single line comments aren't allowed in JSX, so this is definitely an issue. For more context, see https://github.com/eslint/eslint/issues/7030#issuecomment-253602373.

**Comment by kaicataldo** [comment]
This error comes from here: https://github.com/eslint/typescript-eslint-parser/blob/master/lib/convert-comments.js#L113

**Comment by kaicataldo** [comment]
I actually wonder if this is a bug in the TypeScript Scanner? Seems like it only returns this token when it's in `JSXText` nodes.

**Comment by JamesHenry** [comment]
Ported here https://github.com/JamesHenry/typescript-estree/issues/6
